src/test/java/core/basesyntax/services/operations/FruitOperationStrategyAndFruitTransactionProcessorTest.java

```diff
@@ -0,0 +1,102 @@
+package core.basesyntax.services.operations;
+
+import static core.basesyntax.db.Storage.fruitStorage;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exceptions.FruitStorageException;
+import core.basesyntax.services.FruitTransactionProcessor;
+import core.basesyntax.services.impl.FruitTransactionProcessorImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class FruitOperationStrategyAndFruitTransactionProcessorTest {
+
+    private static FruitOperationStrategy strategy;
+    private static List<OperationHandler> handlers;
+    private FruitTransactionProcessor processor;
+
+    @BeforeAll
+    static void beforeAll() {
+        handlers = new ArrayList<>();
+        handlers.add(new BalanceOperationHandler());
+        handlers.add(new PurchaseOperationHandler());
+        handlers.add(new ReturnOperationHandler());
+        handlers.add(new SupplyOperationHandler());
+        strategy = new FruitOperationStrategy(handlers);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        fruitStorage.clear();
+    }
+
+    @Test
+    void getHandlerReturnsCorrectHandlerForBalanceOperationOk() {
+        FruitTransactionDto dto = new FruitTransactionDto("b", "Apple", 10);
+        List<OperationHandler> result = strategy.getHandler(dto);
+        assertEquals(1, result.size(), "Expected one handler");
+        assertTrue(result.get(0) instanceof BalanceOperationHandler,
+                "Expected BalanceOperationHandler");
+    }
+
+    @Test
+    void applyThrowsExceptionWhenFruitNotInStorageNotOk() {
```

Comment:
not Ok means that it throws exception, fix that naming for all tests as it is was pointed out before
```suggestion
    void apply_WhenFruitNotInStorage_NotOk() {
```